kernel/api/src/main/java/org/sakaiproject/messaging/api/MicrosoftMessage.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2003-2017 The Apereo Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *             http://opensource.org/licenses/ecl2
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sakaiproject.messaging.api;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Builder
+@Data
+public class MicrosoftMessage {
+	public static final String CREATE_ELEMENT = "CREATE_ELEMENT";
```

Comment:
I'm fine with this! An Enum would be also good.

public enum MessageType { CREATE_ELEMENT, DELETE_ELEMENT, ADD_MEMBER_TO_AUTHZGROUP , REMOVE_MEMBER_FROM_AUTHZGROUP }

Reply:
Done!